services/user/user.go

```diff
@@ -27,11 +27,52 @@ import (
 // DeleteUser completely and permanently deletes everything of a user,
 // but issues/comments/pulls will be kept and shown as someone has been deleted,
 // unless the user is younger than USER_DELETE_WITH_COMMENTS_MAX_DAYS.
-func DeleteUser(u *user_model.User) error {
+func DeleteUser(ctx context.Context, u *user_model.User, purge bool) error {
 	if u.IsOrganization() {
 		return fmt.Errorf("%s is an organization not a user", u.Name)
 	}
 
+	if purge {
+		// Delete all repos belonging to this user
+		for page := 1; ; page++ {
+			repos, _, err := models.SearchRepositoryByName(&models.SearchRepoOptions{
```

Comment:
So idk if this loop will work, because of pagination and when deleting things it will adjust location of repos when searching, so you'll effectively only delete half of the repos.

Reply:
So we just need always search the first 100 repositories.

Reply:
Agh! @lunny that'd do the trick.

Reply:
Done